tuning/sft_trainer.py

```diff
@@ -339,16 +341,40 @@ def train(
     }
     training_args = SFTConfig(**transformer_kwargs, **additional_args)
 
-    trainer = SFTTrainer(
-        model=model,
-        tokenizer=tokenizer,
-        train_dataset=formatted_train_dataset,
-        eval_dataset=formatted_validation_dataset,
-        data_collator=data_collator,
-        args=training_args,
-        callbacks=trainer_callbacks,
-        peft_config=peft_config,
-    )
+    trainer = None
+    if train_args.enable_reduce_loss_sum:
+        trainer = SumLossSFTTrainer(
+            model=model,
+            tokenizer=tokenizer,
+            train_dataset=formatted_train_dataset,
+            eval_dataset=formatted_validation_dataset,
+            data_collator=data_collator,
+            args=training_args,
+            callbacks=trainer_callbacks,
+            peft_config=peft_config,
+        )
+
+        # Note this check has to be done post trainer initialization
+        if _is_peft_model(trainer.model):
+            logger.warning(
```

Comment:
Shouldn't we rather fail if both enable_reduce_loss_sum and peft flags are enabled?

Reply:
its up to us to decide weather to fail or continue with a user message..if we decide to fail the above flow makes more sense to me.

https://github.com/foundation-model-stack/fms-hf-tuning/pull/520#discussion_r2034832245